backend/iter.go

```diff
@@ -30,15 +32,16 @@ var _ sql.RowIter = (*SQLRowIter)(nil)
 
 // SQLRowIter wraps a standard sql.Rows as a RowIter.
 type SQLRowIter struct {
-	rows      *stdsql.Rows
-	columns   []*stdsql.ColumnType
-	schema    sql.Schema
-	buffer    []any // pre-allocated buffer for scanning values
-	pointers  []any // pointers to the buffer
-	decimals  []int
-	intervals []int
-	nonUTF8   []int
-	charsets  []sql.CharacterSetID
+	rows            *stdsql.Rows
+	columns         []*stdsql.ColumnType
+	schema          sql.Schema
+	buffer          []any // pre-allocated buffer for scanning values
+	pointers        []any // pointers to the buffer
+	decimals        []int
+	intervals       []int
+	nonUTF8         []int
+	charsets        []sql.CharacterSetID
+	typeConversions map[int]reflect.Type
```

Comment:
Is `reflect.Type` necessary? How about the more efficient https://pkg.go.dev/reflect#Kind ? In addition, this field is used in a hot loop, so it may be better to used an array instead of a map. For example:

```
type typeConversion struct {
    idx  int
    kind reflect.Kind
}
```
```
conversions []typeConversion
```